AWSIoT/Internal/SocketRocket/AWSSRWebSocket.m

```diff
@@ -268,22 +269,28 @@ + (void)initialize;
     CRLFCRLF = [[NSData alloc] initWithBytes:"\r\n\r\n" length:4];
 }
 
-- (id)initWithURLRequest:(NSURLRequest *)request protocols:(NSArray *)protocols allowsUntrustedSSLCertificates:(BOOL)allowsUntrustedSSLCertificates;
+- (id)initWithURLRequest:(NSURLRequest *)request protocols:(NSArray *)protocols allowsUntrustedSSLCertificates:(BOOL)allowsUntrustedSSLCertificates hostName:(NSString *)hostName
 {
     if (self = [super init]) {
         assert(request.URL);
         _url = request.URL;
         _urlRequest = request;
         _allowsUntrustedSSLCertificates = allowsUntrustedSSLCertificates;
-        
+        _hostName = hostName ?: _url.host;
```

Comment:
Just to double check. Should we additionally check for the `hostName. length > 0 ` here?